src/core/router/tunnel/config.cc

```diff
@@ -168,17 +239,15 @@ TunnelConfig::TunnelConfig(
   for (auto it : peers) {
     auto hop = new TunnelHopConfig(it);
```

Comment:
As noted above, the lack of smart pointers have always been an issue with this codebase but I/we've been refactoring along the way. This kind of work was out of scope for this branch and it has repercussions elsewhere so, I chose not to touch this for now. Referencing #511.